cpp/src/arrow/compute/kernels/scalar_arithmetic.cc

```diff
@@ -817,24 +818,158 @@ struct Log1pChecked {
   }
 };
 
+struct RoundUtil {
+  template <typename T, enable_if_t<std::is_floating_point<T>::value, bool> = true>
+  static constexpr bool ApproxEqual(const T x, const T y) {
```

Comment:
nit: I think you can write this as

```cpp
template <typename T>
static constexpr enable_if_t<std::is_floating_point<T>::value, bool> ApproxEqual(...)
```

if I'm not mistaken.

Reply:
Yes, you are right. Thanks!